qiskit_nature/results/electronic_structure_result.py

```diff
@@ -136,23 +136,17 @@ def reverse_dipole_sign(self, value: bool) -> None:
     @property
     def total_dipole_moment(self) -> Optional[List[float]]:
         """Returns total dipole of moment"""
-        if self.dipole_moment is None:
-            return None  # No dipole at all
-        tdm: List[float] = []
-        for dip in self.dipole_moment:
-            if np.any(np.equal(list(dip), None)):
-                tdm.append(None)  # One or more components in the dipole is None
-            else:
-                tdm.append(np.sqrt(np.sum(np.power(list(dip), 2))))
-        return tdm
+        if np.all(np.equal(self.dipole_moment[0], None)):
```

Comment:
There are 3 different conditions to take into account here.
1. No dipole moments for X,Y,Z were computed at all
2. Dipole was computed and we have X,Y and Z values returned which can be summed for a total
3. Z2 Symmetry reduction was done, with one or more of the X,Y or Z dipole operators failing to commute with symmetry hence None would be returned for that moment(s), i.e letting caller know its unknown since it could not be validly measured. In this case we have some dipole moments but not enough to compute a total. That logic is reflected in the original code as far as I can see. I do not believe we should report a total when we cannot measure some of them since it can be incorrect which is why we would in the past we would not have shown it

Reply:
Thanks, ok so replacing the np.all() by to np.any() should do the trick. 
Sorry, it is confusing having a None for total dipole, when there is one of its dimensions computed.
Assumed it was another problem.
Perhaps a text-like warning for the users would be good.

Reply:
I would need to look more carefully - I am not sure that caters to when there is no dipole computed at all, maybe that is done outside the method since we should not be trying to compute a total at all in that case - e.g when using the PyQuante driver we have no dipole moment ints and hence we never attempt any dipole computation since we cannot create any dipole operators ( the other cases are when we do that we need to take care of Z2Symmetry, that is done on the main operator, that we can still validly do it on the dipole moments - if not None is the result to indicate the failure). 
How to represent things then in text format that is readable for the end user is a choice. None means you need to understand what that means - but then that block was all formatted so anything longer might be harder to do - if None is acceptable as an individual moment maybe ?, or Unknown, Cannot compute, or something else can be written for a total. Maybe as you suggest we could instead leave None in all places and if None shows up then simply add some text to explain the meaning in the form of say a footnote under the dipole output.

Reply:
@woodsp-ibm I am not sure if you noticed but I reverted this questionable change.